lib/core/src/irods_client_negotiation.cpp

```diff
@@ -32,16 +32,15 @@ namespace irods
 {
     /// =-=-=-=-=-=-=-
     /// @brief given a property map and the target host name decide between a federated key and a local key
-    const std::string& determine_negotiation_key( const std::string& _host_name )
+    const std::string determine_negotiation_key( const std::string& _host_name )
     {
         // search the federation map for the host name
         try {
-            for ( const auto& el : irods::get_server_property<const std::vector<boost::any>>(irods::CFG_FEDERATION_KW) ) {
+            for ( const auto federation : irods::get_server_property<nlohmann::json>(irods::CFG_FEDERATION_KW) ) {
                 try {
-                    const auto& federation = boost::any_cast<const std::unordered_map<std::string, boost::any>&>(el);
                     try {
-                        if ( _host_name == boost::any_cast<const std::string&>(boost::any_cast<const std::vector<boost::any>&>(federation.at(irods::CFG_CATALOG_PROVIDER_HOSTS_KW))[0]) ) {
-                            return boost::any_cast<const std::string&>(federation.at(irods::CFG_NEGOTIATION_KEY_KW));
+                        if (_host_name == federation.at(irods::CFG_CATALOG_PROVIDER_HOSTS_KW)[0].get<std::string>()) {
```

Comment:
Is `operator[]` guaranteed to be non-empty?

Reply:
If not, I would suggest it is misconfigured.  The exception should be updated to match.